extensions/binding/org.eclipse.smarthome.binding.tradfri/src/main/java/org/eclipse/smarthome/binding/tradfri/handler/TradfriLightHandler.java

```diff
@@ -0,0 +1,322 @@
+                        } else {
+                            setColorTemperature(new PercentType(Math.max(current - STEP, PercentType.ZERO.intValue())));
+                        }
+                    } else {
+                        logger.debug("Cannot handle inc/dec as current state is not known.");
+                    }
+                } else {
+                    logger.debug("Can't handle command {} on channel {}", command, channelUID);
+                }
+                break;
+            default:
+                logger.error("Unknown channel UID {}", channelUID);
+        }
+    }
+
+    /**
+     * This class is a Java wrapper for the raw JSON data about the light state.
+     */
+    private static class LightData {
+
+        private final Logger logger = LoggerFactory.getLogger(LightData.class);
+
+        JsonObject root;
+        JsonArray array;
+        JsonObject attributes;
+
+        public LightData() {
+            root = new JsonObject();
+            array = new JsonArray();
+            attributes = new JsonObject();
+            array.add(attributes);
+            root.add(LIGHT, array);
+        }
+
+        public LightData(JsonElement json) {
+            try {
+                root = json.getAsJsonObject();
+                array = root.getAsJsonArray(LIGHT);
+                attributes = array.get(0).getAsJsonObject();
+            } catch (JsonSyntaxException e) {
+                logger.error("JSON error: {}", e.getMessage(), e);
+            }
+        }
+
+        public LightData setBrightness(PercentType brightness) {
+            attributes.add(DIMMER, new JsonPrimitive(Math.round(brightness.floatValue() / 100.0f * 254)));
+            return this;
+        }
+
+        public PercentType getBrightness() {
+            int b = attributes.get(DIMMER).getAsInt();
+            if (b == 1) {
+                return new PercentType(1);
+            }
+            return new PercentType((int) Math.round(b / 2.54));
+        }
+
+        public boolean getReachabilityStatus() {
+            if (attributes.get(REACHABILITY_STATE) != null) {
+                return attributes.get(REACHABILITY_STATE).getAsInt() == 1;
+            } else {
+                return false;
+            }
+        }
+
+        public LightData setTransitionTime(int seconds) {
+            attributes.add(TRANSITION_TIME, new JsonPrimitive(seconds));
+            return this;
+        }
+
+        @SuppressWarnings("unused")
+        public int getTransitionTime() {
+            return attributes.get(TRANSITION_TIME).getAsInt();
+        }
+
+        private final static double[] X = new double[] { 33137.0, 30138.0, 24933.0 };
+        private final static double[] Y = new double[] { 27211.0, 26909.0, 24691.0 };
+
+        public LightData setColorTemperature(PercentType c) {
+            double percent = c.doubleValue();
+
+            long x, y;
```

Comment:
Are they coordinates, otherwise it would be great to given a little bit clearer naming.